packages/backend-defaults/src/entrypoints/auth/authServiceFactory.test.ts

```diff
@@ -450,8 +455,113 @@ describe('authServiceFactory', () => {
         dependencies: [...mockDeps, customPluginTokenHandler],
       });
       const searchAuth = await tester.getSubject('search');
-      searchAuth.authenticate('unlimited-static-token');
+      await searchAuth.authenticate('unlimited-static-token');
       expect(customLogic).toHaveBeenCalledWith('unlimited-static-token');
     });
   });
+  describe('add custom  ExternalTokenHandler', () => {
+    it('should allow custom logic to be injected into the plugin token handler', async () => {
+      const customLogic = jest.fn();
+      const customAddEntry = jest.fn();
+      const customConfig = jest.fn();
+      const deps = [
+        discoveryServiceFactory,
+        mockServices.rootConfig.factory({
+          data: {
+            backend: {
+              baseUrl: 'http://localhost',
+              auth: {
+                keys: [{ secret: 'abc' }],
+                externalAccess: [
+                  {
+                    type: 'static',
+                    options: {
+                      token: 'limited-static-token',
+                      subject: 'limited-static-subject',
+                    },
+                    accessRestrictions: [
+                      { plugin: 'catalog', permission: 'do.it' },
+                    ],
+                  },
+                  {
+                    type: 'static',
+                    options: {
+                      token: 'unlimited-static-token',
+                      subject: 'unlimited-static-subject',
+                    },
+                  },
+                  {
+                    type: 'custom',
+                    options: {
+                      [`custom-config`]: 'custom-config',
+                      foo: 'bar',
+                    },
+                  },
+                ],
+              },
+            },
+          },
+        }),
+      ];
+
+      const customHandler = new (class CustomHandler implements TokenHandler {
+        add(options: Config): TokenHandler {
+          customAddEntry(options);
+          return this;
+        }
+        async verifyToken(token: string): Promise<
+          | {
+              subject: string;
+              allAccessRestrictions?: AccessRestrictionsMap;
+            }
+          | undefined
+        > {
+          customLogic(token);
+          return {
+            subject: 'foo',
+          };
+        }
+      })();
+
+      const customPluginTokenHandler = createServiceFactory({
+        service: externalTokenHandlersServiceRef,
+        deps: {},
+        async factory() {
+          return {
+            custom: (options: Config) => {
+              customConfig(options);
+              return customHandler.add(options);
+            },
```

Comment:
```suggestion
          return {
            type: 'custom',
            factory: (options: Config) => {
              customConfig(options);
              return customHandler.add(options);
            },
```

another pattern to consider